projects/RabbitMQ.Client/client/api/ConnectionFactory.cs

```diff
@@ -647,8 +644,14 @@ private IFrameHandler ConfigureFrameHandler(IFrameHandler fh)
             return fh;
         }
 
-        private void SetUri(Uri uri)
+        private void SetUri(Uri? uri)
         {
+            if (uri is null)
+            {
+                _uri = uri;
+                return;
```

Comment:
In this case, we may also want to reset parameters that are set based on Uri to the default ones. @lukebakken what is your opinion?

Reply:
I would prefer that `_uri` and everything related to it is never null. I'll see if that's possible.

Reply:
@lechu445 @bollhals see my latest commit. Thanks.

Reply:
Assuming the getter of Uri property won't be called in loop by consumers, this change is fine for me.